### PR TITLE
initial import for lifecycle subscriber

### PR DESCRIPTION
**Pull request by Karsten1987**
@wjwwood initial import for lifecycle_subscriber and the respective fixes for the subscription factory.

This is work in progress.

### CODE REVIEW

File: rclcpp/include/rclcpp/subscription_factory.hpp [hunk 1]
```diff
@@ -77,8 +77,7 @@ create_subscription_factory(
   AnySubscriptionCallback<MessageT, Alloc> any_subscription_callback(allocator);
   any_subscription_callback.set(std::forward<CallbackT>(callback));
 
-  auto message_alloc =
-    std::make_shared<typename subscription::Subscription<MessageT, Alloc>::MessageAlloc>();
+  auto message_alloc = std::make_shared<typename subscription::Subscription<MessageT, Alloc>::MessageAlloc>();
```
Comment (Karsten1987) [inline rclcpp/include/rclcpp/subscription_factory.hpp]: Not sure if we have to create a SubscriptionT allocator here

**Comment by mikaelarguedas** [comment]
@Karsten1987 what is the status of this ? Should this stay stale or be closed ?

**Comment by Karsten1987** [comment]
The status is as mentioned in the above comment. This commit compiles, but segfaults. I'll have to debug this in purpose with @wjwwood.
We can put this into the backlog tho.

**Comment by ivanpauno** [comment]
Closing because this has stayed stale too long.
@Karsten1987 if you think that anything should happen here feel free to reopen, open a new PR or a new issue.

**Pull request closed without merging**